### DISCUSSION TITLE
Problem with TNS

### DISCUSSION THREAD

**Discussion by ilyosjon09** [yajra/laravel-oci8#687] [Q&A]
Hello!

I installed the package as per guidelines. I am trying to database using TNS string. Thus I changed config accordingly by showing `tns` key only.

How ever when try to test it throws Undefined index: database.

I tried to connect with oci extension in a custom PHP script and it worked. But I cannot connect using this package. Please help.

FYI:
I am setting `tns` to something like `(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)(HOST=myhost)(PORT=1521)))(CONNECT_DATA=(SERVICE_NAME=servicename)(SERVER=DEDICATED)))`

**Comment by ilyosjon09** [comment]
![image](https://user-images.githubusercontent.com/8179000/142399288-890969a4-a800-450f-999d-4ad1a6120ef9.png)

**Reply by yajra** [reply]
Can you provide an example config that you use? I think you can also use `service_name` and might yield the same TNS you need?

```php
'oracle' => [
    'driver' => 'oracle',
    'host' => 'oracle.host',
    'port' => '1521',
    'database' => 'xe',
    'service_name' => 'sid_alias',
    'username' => 'hr',
    'password' => 'hr',
    'charset' => '',
    'prefix' => '',
]
```

**Reply by ilyosjon09** [reply]
Here's the config:
![image](https://user-images.githubusercontent.com/8179000/142563738-f806cba6-a385-43af-80db-62cf1326a142.png)

In `$tns` variable I am storing TNS string similar to  `(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)(HOST=myhost)(PORT=1521)))(CONNECT_DATA=(SERVICE_NAME=servicename)(SERVER=DEDICATED)))`

I was trying to set up a config similar to the sample from docs:

![image](https://user-images.githubusercontent.com/8179000/142563931-fb0fe61a-e6f4-4f43-a27d-c28609d824ca.png)

**Reply by yajra** [reply]
I think you need to include the database key, TNS will still be prioritize even if it's set.

**Reply by ilyosjon09** [reply]
Ah, okay thank you!

**Comment by aditeleanjar** [comment]
Where i put database key?

**Reply by ilyosjon09** [reply]
Just include it in the config

```diff
[
    'connections' => [
       'oracle' => [
          'driver' => 'oracle',
          'tns' => 'tns',
+       'database' => 'db',
       ]
    ]
]
```
